### PR TITLE
feat(pricing-table): add sticky header

### LINKED ISSUE #8420: [Pricing table] Implement table header fixed scroll behavior
**Engineering info:**  
- Innovation Team engineer: @jkaeser 
- Innovation Team JIRA Ticket: https://jsw.ibm.com/browse/HC-2425
- DPO consulting engineer: @IgnacioBecerra 

-----

Carbon for ibm.com developer

> I need to:
create/change the `Pricing table`

> so that I can:
provide the ibm.com adopter developers components they can use to build ibm.com web pages

#### Additional information
- Story within Storybook with corresponding knobs
- Utilize Carbon
- Create with Shadow DOM and Custom Elements standards
- The React version is created as a wrapper to the Web Components version
- **See the Epic (#8389) for the Design and Functional specs information**
- Visual QA testing issue (#8391)
- Prod QA testing issue (#8392)

#### Acceptance criteria
- [ ] Include README for the web component and corresponding styles
- [ ] Create Web Components styles in styles package
- [ ] Do not create knobs in Storybook that include JSON objects
- [ ] Break out Storybook stories into multiple variation stories, if applicable
- [ ] Create codesandbox example under `/packages/web-components/examples/codesandbox` and include in README
- [ ] Minimum 80% unit test coverage
- [ ] add the trigger to create the React Wrapper
 - [ ] create the React storybook stories and related documentation
- [ ] Update the Carbon for ibm.com website component [data file](https://github.com/carbon-design-system/carbon-for-ibm-dotcom-website/blob/master/src/data/components.json) to be sure the web site Component Status and Storybook links are correct
- [ ] Use the [Visual QA checklist](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/wiki/Definition-of-done-(Visual-QA-checklist)) to verify design quality
- [ ] Update the e2e test scripts, if needed. 
- [ ] Confirm e2e test results have no major errors.
- [ ] A comment is posted in the Prod QA issue (#8392) when development is finished.  
- [ ] The `dev complete` label is added to the Prod QA issue (#8392).

### PR DESCRIPTION
**Pull request by jkaeser**
### Related Ticket(s)

Resolves #8420

### Description

Enhances the `DDSPricingTable` component by making the header sticky on scroll.

### Changelog

**New**

- Adds sticky header to `DDSPricingTable` components.

**Changed**

- Exposes offset value from `StickyHeader` utility class via `height` getter.

### CODE REVIEW

File: packages/web-components/src/components/pricing-table/pricing-table.ts
Comment (IgnacioBecerra) [inline packages/web-components/src/components/pricing-table/pricing-table.ts:288]: It's all looking good! Maybe we could have a constant for the `-10px` instead?

**Comment by RichKummer** [comment]
@jkaeser just a heads up, having a side conversation with Jen about the fixed version styling

File: packages/web-components/src/components/pricing-table/pricing-table.ts
Comment (jkaeser) [reply packages/web-components/src/components/pricing-table/pricing-table.ts]: Thanks for the review, @IgnacioBecerra! I set this value in a `_startSentinelBuffer` class property.

Let me know if you want to take a look at the rest of this together. There's a lot to make sense of, and I'd be happy to walk you through it all.

**Comment by RichKummer** [comment]
> Thanks for the review, @RichKummer!
> Darn, I thought I solved this one. Do you know what screen size and browser you saw this issue on?

It was at the `xlg` breakpoint, using Chrome - I'm trying to see if I can replicate it again

**Comment by RichKummer** [comment]
> Darn, I thought I solved this one. Do you know what screen size and browser you saw this issue on?

@jkaeser 
It was at the `xlg` breakpoint, using Chrome. It looks like it's showing up on `lg` and up, but not on `md` or `sm`. It seems like the root might be the window height - it was bouncing at `778px` height, but not `6--px` height.

Uploading pricing-table-02.mov…

**Comment by RichKummer** [comment]
Rerencing comment above:
https://user-images.githubusercontent.com/15643582/169105643-d1ef8a64-f568-4207-b3b8-7539a1211c59.mov

**Comment by jkaeser** [comment]
> Rerencing comment above: https://user-images.githubusercontent.com/15643582/169105643-d1ef8a64-f568-4207-b3b8-7539a1211c59.mov

@RichKummer can you take another look? Should be fixed now.

**Comment by jwitkowski79** [comment]
@jkaeser @RichKummer @oliviaflory Hello! We've resolved the design of the sticky table header and are ready to move forward with this. Details below:

- When the header is in the sticky state, the table header should expand full width and the border-bottom should be removed. 
- Updated the box-shadow to match the Carbon box-shadow

Screen Shot 2022-06-03 at 9 18 10 AM


Updated design specs: [https://ibm.box.com/s/ydxpyl9unhd9x572d4l9hzkva3vomcu0](https://ibm.box.com/s/ydxpyl9unhd9x572d4l9hzkva3vomcu0)

**Comment by jkaeser** [comment]
> - When the header is in the sticky state, the table header should expand full width and the border-bottom should be removed.
> - Updated the box-shadow to match the Carbon box-shadow

@jwitkowski79 @RichKummer I updated the fixed header according to the new specifications. Please take a look when you have chance. Thanks! 

https://ibmdotcom-web-components-experimental.s3.us-east.cloud-object-storage.appdomain.cloud/deploy-previews/8824/index.html?path=/story/components-pricing-table--with-subheaders

**Comment by RichKummer** [comment]
@jkaeser - I took a look at the fixed header, it looks like the new styling has been incorporated for the removed horizontal rule and the Carbon box-shadow. I'll let @jwitkowski79 take a second look though.

Percy did catch a few things - I'm wondering if you just need to merge some changes from `main`? The build seems to rollback an update from #8935 shifting the ToC over to the left:
Screen Shot 2022-06-13 at 4 50 51 PM

**Comment by jwitkowski79** [comment]
@jkaeser This looks good! The only thing I was wondering is if we can smooth out the animation a bit more when the fixed sticky header drops in? I think it might need more of an ease in. Let me know if you want to hop into a call to review and discuss!

**Comment by jkaeser** [comment]
> @jkaeser This looks good! The only thing I was wondering is if we can smooth out the animation a bit more when the fixed sticky header drops in? I think it might need more of an ease in. Let me know if you want to hop into a call to review and discuss!

@jwitkowski79 I added an animation to make the header slide in from the top when it becomes sticky. Let me know if this is along the lines of what you were thinking. Thanks!

**Comment by jwitkowski79** [comment]
@jkaeser This looks much better! Thanks!!

**Review by jwitkowski79 — APPROVED** [review]

**Review by proeung — APPROVED** [review]
LGTM!

---------

@kennylam or @IgnacioBecerra Now that we got the design sign-off on this PR. Can we get your review so that this PR can be marked as "Ready to merge"?

**Review by IgnacioBecerra — APPROVED** [review]
LGTM, works great!! Thanks @jkaeser!

**Pull request merged**
